logicalplan/histogram_stats.go

```diff
@@ -0,0 +1,29 @@
+package logicalplan
+
+import (
+	"github.com/prometheus/prometheus/util/annotations"
+
+	"github.com/thanos-io/promql-engine/query"
+)
+
+type DetectHistogramStatsOptimizer struct{}
+
+func (d DetectHistogramStatsOptimizer) Optimize(plan Node, _ *query.Options) (Node, annotations.Annotations) {
+	var decodeStats bool
+	Traverse(&plan, func(node *Node) {
+		switch n := (*node).(type) {
+		case *VectorSelector:
+			n.DecodeNativeHistogramStats = decodeStats
```

Comment:
Isnt it better to be explicit here and set as false?

Reply:
What do you mean by this? Set what to false?

Reply:
nvm, just realized we are traversing 🙂